packages/discovery-react-components/src/components/ResultsPagination/ResultsPagination.tsx

```diff
@@ -63,7 +82,49 @@ const ResultsPagination: FC<ResultsPaginationProps> = ({
     isResultsPaginationComponentHidden
   } = useContext(SearchContext);
   const [currentPage, setCurrentPage] = useState(page);
-  const resultsPerPage = componentSettings?.results_per_page ?? 10;
+  const [resetCounter, setResetCounter] = useState(0);
+  const resultsPerPage = pageSize || searchParameters.count || 10;
+
+  const handleOnChange = useCallback(
+    (evt: ResultsPaginationEvent): void => {
+      if (onChange) {
+        onChange(evt);
+      }
+      const { page, pageSize } = evt;
+      const offset = (page - 1) * pageSize;
+      setCurrentPage(page);
+      performSearch(
+        {
+          ...searchParameters,
+          count: pageSize,
+          offset
+        },
+        false
+      );
+    },
+    [onChange, performSearch, searchParameters]
+  );
+
+  // Since the Carbon pagination component controls the page number via its state
+  // without a way to influence it via props, we occasionally need to reset it back to "Page 1".
+  // This is done via a key on the CarbonPagination element, which is triggered via a counter
+  // that is incremented by this function.
+  const reset = useCallback(
+    ({ triggerOnChange }: ResetOptions) => {
+      setCurrentPage(1);
+      // This is structured as a counter so that the `key` prop of the Carbon pagination component
+      // updates (triggering a full re-render) in a simple and performant manner
+      setResetCounter(resetCounter + 1);
+      if (triggerOnChange) {
+        handleOnChange({ page: 1, pageSize: resultsPerPage });
+      }
+    },
+    [handleOnChange, resetCounter, resultsPerPage]
+  );
+
+  // Externalize the reset function to a ref that the parent can send in,
+  // so that it can also reset the pagination as desired
+  useImperativeHandle(resetRef, () => ({ reset }));
```

Comment:
There should be a mention of the available functions (just `reset` currently) that are available to call directly, in the JSDoc for this component.